examples/timestamp/signed_timestamp.c

```diff
@@ -113,31 +120,36 @@ int TPM2_Timestamp_Test(void* userCtx)
 
     /* Create RSA Storage Key, also called SRK */
     /* See if SRK already exists */
-    rc = wolfTPM2_ReadPublicKey(&dev, &storage, TPM2_DEMO_STORAGE_KEY_HANDLE);
+    if (tryNVkey) {
+        rc = wolfTPM2_ReadPublicKey(&dev, &storage, TPM2_DEMO_STORAGE_KEY_HANDLE);
 #ifdef TEST_WRAP_DELETE_KEY
-    if (rc == 0) {
-        storage.handle.hndl = TPM2_DEMO_STORAGE_KEY_HANDLE;
-        rc = wolfTPM2_NVDeleteKey(&dev, TPM_RH_OWNER, &storage);
-        if (rc != 0) goto exit;
-        rc = TPM_RC_HANDLE; /* mark handle as missing */
-    }
+        if (rc == 0) {
+            storage.handle.hndl = TPM2_DEMO_STORAGE_KEY_HANDLE;
+            rc = wolfTPM2_NVDeleteKey(&dev, TPM_RH_OWNER, &storage);
+            if (rc != 0) goto exit;
+            rc = TPM_RC_HANDLE; /* mark handle as missing */
+        }
 #endif
-    if (rc != 0) {
+    }
+
+    if (!tryNVkey || (tryNVkey && rc != 0)) {
```

Comment:
maybe we need to document(comment) the tryNVkey variable or this if check here. Took me few seconds to realize this solves the NV persistence issue, but I have context. I wonder if someone without context will understand the idea behind tryNVkey and the rule. Just thinking out loud. This is not a change request.

Reply:
@elms , I agree about adding a comment here for what `tryNVkey` is. I had the same thought during my review. Had to double take for what its doing.

Also this flow for `tryNVkey` should be added to the other examples. If you want to follow this up with a new PR that is okay with me.